File: app/src/main/java/edu/osu/table/Wireless_Database.java
package edu.osu.table;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

@Database(entities = {Wireless_Data.class},version = 1)
public abstract class Wireless_Database extends RoomDatabase
{
    public abstract Wireless_Dao wirelessDao();

}

File: app/src/main/java/edu/osu/table/ui/WirelessDataFolder/WirelessDatabase.kt
package edu.osu.table.ui.WirelessDataFolder

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context

@Database(entities = arrayOf(WirelessData::class), version = 1)
abstract class WirelessDatabase : RoomDatabase() {

    abstract fun wirelessDataDao(): WirelessDao

    companion object {
        private var INSTANCE: WirelessDatabase? = null

        fun getInstance(context: Context): WirelessDatabase? {
            if (INSTANCE == null) {
                synchronized(WirelessDatabase::class) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            WirelessDatabase::class.java, "wirelessdb.db").allowMainThreadQueries()
                            .build()
                }
            }
            return INSTANCE
        }

        fun destroyInstance() {
            INSTANCE = null
        }
    }
}